### PR TITLE
[android][image-picker] Support for cameraType

### PR DESCRIPTION
**Pull request by YoussefHenna**
# Why
- Support for `cameraType` option was missing for android, and only implemented in iOS.
- Documentation and types for `cameraType` were missing

# How
- Implemented into Android by adding `Intent` extras that tell the camera app which camera to use by default
  - This of course is dependent on support from the camera app. I send out all known (that I could find) extras that a camera app might support and hope for the best.
- Added types and docs for `CameraType` similarly to how expo-camera currently implements it for consistency. (Maybe should be a shared type?)

# Test Plan
- Ran the `bare-expo` app on Android
- Imported `expo-image-picker` and launched the camera with different values for `cameraType`
- Logged the intent extras to ensure intent extras properly sent.
- It should be noted that this had no effect on my device (Samsung S22 Ultra), most likely my camera app does not support this.

# Checklist
- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CameraContract.kt [hunk 1]
```diff
@@ -34,6 +35,15 @@ internal class CameraContract(
...
           putExtra(MediaStore.EXTRA_DURATION_LIMIT, input.options.videoMaxDuration)
         }
+        if (input.options.cameraType == CameraType.FRONT){
+          putExtra("android.intent.extras.LENS_FACING_FRONT", 1)
+          putExtra("android.intent.extras.CAMERA_FACING", 1)
+          putExtra("android.intent.extra.USE_FRONT_CAMERA", true)
+        } else {
+          putExtra("android.intent.extras.LENS_FACING_BACK", 1)
+          putExtra( "android.intent.extras.CAMERA_FACING", 0)
+          putExtra( "android.intent.extra.USE_FRONT_CAMERA", false)
```
Comment (lukmccall) [inline packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CameraContract.kt:45]:
```suggestion
          putExtra("android.intent.extras.CAMERA_FACING", 0)
          putExtra("android.intent.extra.USE_FRONT_CAMERA", false)
```

File: packages/expo-image-picker/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 ### 🎉 New features
 
+- Added support for the `cameraType` option for Android + Updated types to include the `cameraType` option in `ImagePickerOptions` ([#22143](https://github.com/expo/expo/pull/22143) by [@YoussefHenna](https://github.com/YoussefHenna))
```
Comment (lukmccall) [inline packages/expo-image-picker/CHANGELOG.md:9]:
```suggestion
- Added ability to choose the camera-facing type. ([#22143](https://github.com/expo/expo/pull/22143) by [@YoussefHenna](https://github.com/YoussefHenna))
```
That option wasn't documented, so I think we should treat that as it wasn't added before on iOS.

**Review by lukmccall — CHANGES_REQUESTED** [review]
Thanks for your contribution 🏅

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CameraContract.kt (hunk 1 above)
Comment (YoussefHenna) [reply packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CameraContract.kt:45]: Nice catch!

File: packages/expo-image-picker/src/ImagePicker.types.ts [hunk 3]
```diff
@@ -420,6 +431,15 @@ export type ImagePickerOptions = {
    * @platform ios
    */
   presentationStyle?: UIImagePickerPresentationStyle;
+  /**
+   * Camera facing. Use one of `CameraType`. When `CameraType.front`, use the front-facing camera.
+   * When `CameraType.back`, use the back-facing camera.
+   * - **On Android**, effect of this option depends on support of installed camera app.
```
[the remark this answers is not in the document]
Comment (YoussefHenna) [reply packages/expo-image-picker/src/ImagePicker.types.ts:437]: This was copied from expo-camera's description for `cameraType`, may be worth updating there as well?

**Comment by lukmccall** [comment]
Please fix `Android Unit Tests` job by running `./gradlew :expo-image-picker:spotlessApply` in `bare-expo/android` directory

**Comment by YoussefHenna** [comment]
@lukmccall You can retry the test now and all should be good.

**Review by lukmccall — APPROVED** [review]

**Pull request merged**